tracex/extraction/logic/utils.py

```diff
@@ -201,3 +201,14 @@ def create_dfg_from_df(df):
         os.remove(temp_file_path)
         dfg_img_base64 = base64.b64encode(dfg_img_buffer.getvalue()).decode("utf-8")
         return dfg_img_base64
+
+    @staticmethod
+    def align_df_datatypes(source_df, target_df):
+        """Aligns the datatypes of two dataframes."""
+        for column in source_df.columns:
+            if column in target_df.columns and "datetime" not in str(source_df[column].dtype):
+                source_df[column] = source_df[column].astype(target_df[column].dtype)
+            elif "datetime" in str(source_df[column].dtype):
```

Comment:
I'm not really sure why we cast the type to string and compare with "datetime" instead of checking if the column has a certain type directly.

Reply:
The reason was that I wanted to write this versatile so any timezone (if they occur) are going to be removed. It's true that the string checking is not the best solution. I found the is_datetime64_any_dtype function from pandas.api.types which I'm using now.